api/python/provisioner/commands/configure_setup.py

```diff
@@ -90,199 +98,166 @@ def __attrs_post_init__(self):
-
-
-@attr.s(auto_attribs=True)
-class StorageEnclosureParamsValidation:
-    type: str = StorageEnclosureParams.type
-    primary_ip: str = StorageEnclosureParams.primary_ip
-    secondary_ip: str = StorageEnclosureParams.secondary_ip
-    controller_user: str = StorageEnclosureParams.controller_user
-    controller_secret: str = StorageEnclosureParams.controller_secret
-    controller_type: str = StorageEnclosureParams.controller_type
-    _optional_param = [
-        'controller_type'
-    ]
-
-    def __attrs_post_init__(self):
-        params = attr.asdict(self)
-        # FIXME why we allow any params for the following types?
-        types = ['JBOD', 'virtual', 'RBOD', 'other']
-        if params['type'] in types:
-            return
-        missing_params = []
-        for param, value in params.items():
-            if value == UNCHANGED and param not in self._optional_param:
-                missing_params.append(param)
-        if len(missing_params) > 0:
-            raise ValueError(f"Mandatory param missing {missing_params}")
-
-
-@attr.s(auto_attribs=True)
-class NodeParamsValidation:
-    hostname: str = NodeNetworkParams.hostname
-    roles: List = NodeNetworkParams.roles
-    data_public_interfaces: List = NodeNetworkParams.data_public_interfaces
-    data_private_interfaces: List = NodeNetworkParams.data_private_interfaces
-    data_public_ip: str = NodeNetworkParams.data_public_ip
-    data_netmask: str = NodeNetworkParams.data_netmask
-    data_gateway: str = NodeNetworkParams.data_gateway
-    mgmt_interfaces: List = NodeNetworkParams.mgmt_interfaces
-    mgmt_public_ip: str = NodeNetworkParams.mgmt_public_ip
-    mgmt_netmask: str = NodeNetworkParams.mgmt_netmask
-    mgmt_gateway: str = NodeNetworkParams.mgmt_gateway
-    data_private_ip: str = NodeNetworkParams.data_private_ip
-    bmc_user: str = NodeNetworkParams.bmc_user
-    bmc_secret: str = NodeNetworkParams.bmc_secret
-
-    _optional_param = [
-        'data_public_ip',
-        'roles',
-        'data_netmask',
-        'data_gateway',
-        'data_private_ip',
-        'mgmt_interfaces',
-        'mgmt_public_ip',
-        'mgmt_netmask',
-        'mgmt_gateway'
-    ]
-
-    def __attrs_post_init__(self):
-        params = attr.asdict(self)
-        missing_params = []
-        for param, value in params.items():
-            if value == UNCHANGED and param not in self._optional_param:
-                missing_params.append(param)
-        if len(missing_params) > 0:
-            raise ValueError(f"Mandatory param missing {missing_params}")
-
-
 @attr.s(auto_attribs=True)
 class ConfigureSetup(CommandParserFillerMixin):
     input_type: Type[inputs.NoParams] = inputs.NoParams
     _run_args_type = RunArgsConfigureSetup
 
-    validate_map = {
-        "cluster": NetworkParamsValidation,
-        "node": NodeParamsValidation,
-        "storage": StorageEnclosureParamsValidation
-    }
+    validate_map = {f'{CLUSTER}': ClusterParamsValidation,
```

Comment:
could you add a simple comment about what are the keys (section name of the ini) and values (validation class).

Reply:
Added a comment for user to understand the param scope.